src/mon/OSDMonitor.cc

```diff
@@ -3106,7 +3106,7 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
     int64_t epochnum;
     cmd_getval(g_ceph_context, cmdmap, "epoch", epochnum, (int64_t)0);
     epoch = epochnum;
-    if (!epoch)
+    if (epoch > 0)
```

Comment:
the default value of `epoch` is "0" if it's not specified. we use it to tell if user wants to get the latest osdmap.

with your change, if user puts `ceph osd getmap`. he/she will get an error.
```
$ ./bin/ceph osd getmap
Error ENOENT: there is no map for epoch 0
```